src/sequences/func.py

```diff
@@ -313,13 +314,15 @@ def remove_duplicate_lists(input_list):
     """
     seen = set()
     result = []
+    hash_values = set()
 
     for sublist in input_list:
-        # Convert each sublist to a tuple to make it hashable
-        sublist_tuple = tuple(sublist)
+        # Calculate a hash value for the sublist
```

Comment:
Using hashes boosts the search process.
However, there are possible problems with this approach, e.g. un-hashable types as inner list elements.